### PR TITLE
upgrade Python to v 3.8

### PR DESCRIPTION
**Pull request by reinvantveer**
**Description of the change:**
Upgrade Python to version 3.8 to facilitate local development (i.e. without having to resort to deadsnakes repos and suchlikes) and benefit from performance enhancements in more recent versions of Python (with 3.6 functionally mothballed now)

**Motivation for the change:**
The Ansible operator uses an outdated version of Python that hampers local development and reproducible builds as it requires you to keep around an old version of Python. Since many recent distros don't ship Python 3.6 in their standard packaging repositories anymore, this requires a manual install, or alternate package repositories.

*Note*: this is a first pull request to start addressing the issues mentioned in #4237 

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by reinvantveer** [comment]
Re-committed to fix DCO signoffs

File: website/content/en/docs/building-operators/ansible/installation.md [hunk 1]
```diff
@@ -9,14 +9,15 @@ weight: 1
 Follow the steps in the [installation guide][install-guide] to learn how to install the `operator-sdk` CLI tool.
 
 ## Additional Prerequisites
-
+- [python][python] version 3.8.6+
```
Comment (estroz) [inline website/content/en/docs/building-operators/ansible/installation.md:12]: Are there breaking changes and/or features between 3.6 and 3.8 that would require users to use 3.8+?

**Comment by dalbani** [comment]
And what about the use of `latest` in the base image `FROM registry.access.redhat.com/ubi8/ubi:latest`?
Instead of pinning to a specific version I mean, for reproducibility purposes.
Tag `8.3-227` is at the moment the latest release: https://catalog.redhat.com/software/containers/ubi8/ubi/5c359854d70cc534b3a3784e.

File: website/content/en/docs/building-operators/ansible/installation.md (hunk 1 above)
Comment (reinvantveer) [reply website/content/en/docs/building-operators/ansible/installation.md:12]: The main feature here is that 3.8+ supports a local dev environment and lots of security and performance updates. In this way, it eases local development and allows you to use Python modules that use 3.7+ language features described in https://docs.python.org/3/whatsnew/3.7.html and https://docs.python.org/3/whatsnew/3.8.html.

To be clear: we rely heavily on Ansible custom modules (i.e. Python modules) to fine-tune our Ansible operators. As such, we require it for our development process.

Comment (reinvantveer) [reply website/content/en/docs/building-operators/ansible/installation.md:12]: Building the Ansible operator locally with Python versions older than 3.8 will most probably work, but the point is reproducibility: a local development environment that mirrors the one you build in the Dockerfile

Comment (reinvantveer) [reply website/content/en/docs/building-operators/ansible/installation.md:12]: Does this answer your question?

Comment (estroz) [reply website/content/en/docs/building-operators/ansible/installation.md:12]: Yes thanks!

**Comment by estroz** [comment]
@dalbani that's a reasonable request. If ansible's base image is being pinned, helm's should as well. This should be done in a separate PR though.

**Review by jmrodri — APPROVED** [review]
/lgtm

**Comment by reinvantveer** [comment]
Great, thanks

**Pull request merged**